Registration/RegisterWithAzure.psm1

```diff
@@ -1651,54 +1647,41 @@ function Get-AzToken
         {
-            $m = 'Please ensure you have authenticated with Az Accounts module!'
-            $m += ' Unable to resolve authentication client factory from Az Accounts module runtime'
-            $m += ' ([Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory])'
-            Write-Error $m
-            return
-        }
-
-        $client = $authenticationClientFactory.CreatePublicClient(
-            ($clientId='1950a258-227b-4e31-a9cf-717495945fc2'),
-            ($TenantId),
-            ($authority="$($Context.Environment.ActiveDirectoryAuthority.TrimEnd('/'))/$TenantId"),
-            ($redirectUri='urn:ietf:wg:oauth:2.0:oob'),
-            ($useAdfs=$Context.Environment.ActiveDirectoryAuthority -like '*/adfs*'))
-
-        $authenticationClientFactory.RegisterCache($client)
+            [Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory]$authenticationClientFactory = $null
+            if (-not ([Microsoft.Azure.Commands.Common.Authentication.AzureSession]::Instance.TryGetComponent(
+                [Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory]::AuthenticationClientFactoryKey,
+                [ref]$authenticationClientFactory)))
+            {
+                $m = 'Please ensure you have authenticated with Az Accounts module!'
+                $m += ' Unable to resolve authentication client factory from Az Accounts module runtime'
+                $m += ' ([Microsoft.Azure.Commands.Common.Authentication.Authentication.Clients.AuthenticationClientFactory])'
+                Write-Error $m
+                return
+            }
 
-        $accounts = $client.GetAccountsAsync().ConfigureAwait($true).GetAwaiter().GetResult()
+            $client = $authenticationClientFactory.CreatePublicClient(
+                ($clientId='1950a258-227b-4e31-a9cf-717495945fc2'),
+                ($TenantId),
+                ($authority="$($Context.Environment.ActiveDirectoryAuthority.TrimEnd('/'))/$TenantId"),
+                ($redirectUri='urn:ietf:wg:oauth:2.0:oob'),
+                ($useAdfs=$Context.Environment.ActiveDirectoryAuthority -like '*/adfs*'))
 
-        $bytes = ([Microsoft.Identity.Client.ITokenCacheSerializer]$client.UserTokenCache).SerializeMsalV3()
+            $authenticationClientFactory.RegisterCache($client)
+            $accounts = $client.GetAccountsAsync().ConfigureAwait($true).GetAwaiter().GetResult()
+            $bytes = ([Microsoft.Identity.Client.ITokenCacheSerializer]$client.UserTokenCache).SerializeMsalV3()
+        }
+        else
+        {
+            $provider = [Microsoft.Azure.Commands.Common.Authentication.SharedTokenCacheProvider]::new()
+            $accounts = $provider.ListAccounts()
+            $bytes = $provider.ReadTokenData()
+        }
         $json  = [System.Text.Encoding]::UTF8.GetString($bytes)
```

Comment:
let's maybe add a check here for $bytes.Length and give some error message if its empty?

Reply:
updated